spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/AggregateApplicationBuilder.java

```diff
@@ -174,20 +135,40 @@ public ProcessorConfigurer via(Class<?> processor) {
 
 		String[] profiles = null;
 
-		ConfigurableApplicationContext parentContext;
-
 		String namespace;
 
 		Class<?> getApp() {
 			return this.app;
 		}
 
-		public void setParentContext(ConfigurableApplicationContext parentContext) {
-			this.parentContext = parentContext;
+		public T as(String... names) {
+			this.names = names;
+			return getConfigurer();
+		}
+
+		public T args(String... args) {
+			this.args = args;
+			return getConfigurer();
+		}
+
+		public T profiles(String... profiles) {
+			this.profiles = profiles;
+			return getConfigurer();
+		}
+
+		public T parent(ConfigurableApplicationContext parentContext) {
+			AggregateApplicationBuilder.this.parentContext = parentContext;
+			return getConfigurer();
```

Comment:
We can register a `SharedChannelRegistry` here if not already defined. Or we can generalize the problem and support shared channels as via `@EnableBinding` and use a `@ConditionalOnMissingBean` with parent lookup for preventing them to be created in the children.